src/java/org/pantsbuild/tools/junit/impl/ConsoleRunnerImpl.java

```diff
@@ -411,12 +420,25 @@ public FailFastRunner(Runner wrappedRunner) {
     this.useExperimentalRunner = useExperimentalRunner;
   }
 
+  private void setTestsToRun(Collection<String> tests) {
+    this.testsToRun = tests;
+  }
+
   void run(Collection<String> tests) {
+    setTestsToRun(tests);
+    run();
+  }
+
+  @VisibleForTesting
+  public void run() {
     System.setOut(new PrintStream(swappableOut));
     System.setErr(new PrintStream(swappableErr));
 
     JUnitCore core = new JUnitCore();
 
+     // number of hooks: 1 consoleShutdownListener + 1 xmlShutdownListener if writing XML output
```

Comment:
Rather than documenting this, it would be good to compute it from the number of hooks that have been installed, and/or increment it when hooks are installed.